addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/XiaomiHandlerFactory.java

```diff
@@ -84,8 +94,22 @@ protected ThingHandler createHandler(Thing thing) {
             XiaomiBridgeHandler handler = new XiaomiBridgeHandler((Bridge) thing);
             registerItemDiscoveryService(handler);
             return handler;
-        } else if (XiaomiItemHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
-            return new XiaomiItemHandler(thing);
+        } else if (XiaomiDeviceBaseHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
```

Comment:
this one probably is not needed i guess. since each if/elseif inside does this check now. correct?